dns-rc-autoscaler/nanny/scaler.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nanny
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+// Scaler determines the number of replicas to run
+type Scaler struct {
+	ConfigFile string
+	Verbose    bool
+}
+
+const paramsSeparator = "nodes.gte."
+
+func (s Scaler) scaleWithNodes(numCurrentNodes uint64) uint64 {
+	newMap, err := ParseScalerParamsFile(s.ConfigFile)
+	if err != nil {
+		log.Fatalf("Parse failure: The configmap volume file is malformed (%s)\n", err)
+	}
+
+	// construct a search ladder from the map
```

Comment:
Yes, but he's doing a sort (so he doesn't need to keep track of a max). I'm fairly ambivalent about linear scan vs binary search when n ~ 2k, thus the conditionals.